config/app/api/openapi/registry.yml

```diff
@@ -0,0 +1,117 @@
+openapi: 3.0.0
+info:
+    version: 1.0.0
+    contact:
+        name: Spryker
+        url: 'https://apps.spryker.com/'
+        email: support@spryker.com
+    title: 'Spryker API'
+    license:
+        name: MIT
+paths:
+    '/private/application-config':
+        post: &paths/private/application-config/post
+            summary: 'Saves or updates application configuration between Tenant and App.'
+            parameters:
+                -   name: X-Store-Reference
+                    in: header
+                    required: true
+                    description: 'Reference of the Tenants store.'
+                    schema:
+                        type: string
+            requestBody:
+                description: 'Provides configuration for the application.'
+                required: true
+                content:
+                    application/json:
+                        schema:
+                            $ref: '#/components/schemas/ConfigurationApiRequest'
+            responses:
+                200:
+                    description: 'Expected response to a valid request.'
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/ConfigurationApiResponse'
+                400:
+                    description: 'Bad request.'
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/ApiErrorMessages'
+                default:
+                    description: 'Expected response to a bad request.'
+                    content:
+                        text/plain:
+                            schema:
+                                type: string
+        put: *paths/private/application-config/post # $ref is not (officially) supported for operation objects
```

Comment:
@k4emic While I'm doing contract tests I found out that this is not working. Do we have any other option than just duplicating the whole operation?

Reply:
We can reference the re-usable parts, here's the smallest sample I was able to come up with using `$ref`:
```
put: 
  summary: 'Updates application configuration between Tenant and App'
  parameters:
    - $ref ...
  requestBody:
    $ref: '#/paths/~1private~1application-config/post/requestBody'
  responses:
    200:
      $ref: '#/paths/~1private~1application-config/post/responses/200'
    400:
      $ref: '#/paths/~1private~1application-config/post/responses/400'
    default:
      $ref: '#/paths/~1private~1application-config/post/responses/default'
```

This can be made less fugly by moving more of the re-usable parts into the [components](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#components-object) section, as this can contain responses, parameters and request bodies as well.

I actually think doing this is a good idea, since it means that loading and re-saving the YML file from a program won't throw away and anchor tag and inflate the entire thing.

Reply:
moved everything reusable to components